nixos/modules/services/x11/window-managers/awesome.nix

```diff
@@ -6,7 +6,8 @@ let
 
   cfg = config.services.xserver.windowManager.awesome;
   awesome = cfg.package;
-
+  inherit (pkgs.luaPackages) getLuaPath getLuaCPath;
+  joinLuaEnv = type: modules: pkgs.stdenv.lib.concatStringsSep ";" (map type modules);
```

Comment:
Use `lib` directly. (without `pkgs.stdenv`).

I also would inline this function.